src/python/gudhi/sklearn/cover_complex.py

```diff
@@ -0,0 +1,473 @@
+        cover_name (string): name of cover. Used when generating plots.
+        color_name (string): name of color function. Used when generating plots.
+        verbose (bool): whether to display info while computing.
+ 
+        simplex_tree (gudhi SimplexTree): simplicial complex representing the cover complex computed after calling the fit() method.
+        node_info_ (dictionary): various information associated to the nodes of the cover complex. 
+    """
+    def __init__(self, complex_type="mapper", input_type="point cloud", cover="functional", colors=None, mask=0,
+                       voronoi_samples=100, assignments=None, filters=None, filter_bnds=None, resolutions=None, gains=None, N=100, beta=0., C=10.,
+                       clustering=None,
+                       graph="rips", rips_threshold=None, 
+                       distance_matrix_name="", input_name="data", cover_name="cover", color_name="color", verbose=False):
+
+        self.complex_type, self.input_type, self.cover, self.colors, self.mask = complex_type, input_type, cover, colors, mask
+        self.voronoi_samples, self.assignments, self.filters, self.filter_bnds, self.resolutions, self.gains, self.clustering = voronoi_samples, assignments, filters, filter_bnds, resolutions, gains, clustering
+        self.graph, self.rips_threshold, self.N, self.beta, self.C = graph, rips_threshold, N, beta, C
+        self.distance_matrix_name, self.input_name, self.cover_name, self.color_name, self.verbose = distance_matrix_name, input_name, cover_name, color_name, verbose
+
+    def fit(self, X, y=None):
+        """
+        Fit the CoverComplex class on a point cloud or a distance matrix: compute the cover complex and store it in a simplex tree called simplex_tree.
+
+        Parameters:
+            X (numpy array of shape (num_points) x (num_coordinates) if point cloud and (num_points) x (num_points) if distance matrix): input point cloud or distance matrix.
+            y (n x 1 array): point labels (unused).
+        """
+        self.data = X
+
+        if self.colors is None:
+            if self.input_type == "point cloud":
+                self.colors = X[:,0] if self.complex_type == "gic" else X[:,0:1]
+            elif self.input_type == "distance matrix":
+                self.colors = X.max(axis=0) if self.complex_type == "gic" else X.max(axis=0)[:,np.newaxis]
+
+        if self.filters is None:
+            if self.input_type == "point cloud":
+                self.filters = X[:,0] if self.complex_type == "gic" else X[:,0:1]
+            elif self.input_type == "distance matrix":
+                self.filters = X.max(axis=0) if self.complex_type == "gic" else X.max(axis=0)[:,np.newaxis]
+
+        if self.complex_type == "gic" or self.complex_type == "nerve":
+
+            if self.complex_type is not "nerve" or self.cover is not "functional": # Nerve + functional cover is a special case where it's better to use Mapper than C++ code
```

Comment:
`is not` should only be used for `None`, for instance. Use `!=` operator for literals.
```suggestion
            if self.complex_type != "nerve" or self.cover != "functional": # Nerve + functional cover is a special case where it's better to use Mapper than C++ code
```

Reply:
Thanks! Done.